onnxruntime/contrib_ops/cuda/transformers/beam_search.cc

```diff
@@ -48,6 +46,10 @@ BeamSearch::BeamSearch(const OpKernelInfo& info)
                    GenerationCudaDeviceHelper::InitBeamState<float>,
                    GenerationCudaDeviceHelper::InitBeamState<MLFloat16>);
 
+#ifndef USE_ROCM
```

Comment:
Should this be `#ifdef USE_CUDA` instead? The `SetDeviceHelpers_Cuda` method in `onnxruntime/contrib_ops/cpu/transformers/beam_search.h` is using `#ifdef USE_CUDA`:

```
#ifdef USE_CUDA
  void SetDeviceHelpers_Cuda(
      const GenerationDeviceHelper::ReorderPastStateFunc& reorder_past_state_func,
      const GenerationDeviceHelper::InitCacheIndirFunc& init_cache_indir_func) {
    reorder_past_state_func_ = reorder_past_state_func;
    init_cache_indir_func_ = init_cache_indir_func;
  }
#endif
```